resolwe/flow/executors/init_container.py

```diff
@@ -0,0 +1,210 @@
+"""Kubernetes init container initialization script.
+
+1. Fix permission (make genialis user owner) on subdirectories on the EBS
+   volume used for local processing.
+2. Transfer missing data to the input EBS volume.
+"""
+
+import asyncio
+import logging
+import os
+import shutil
+import sys
+from contextlib import suppress
+from distutils.util import strtobool
+from pathlib import Path
+from typing import Optional
+
+import zmq
+import zmq.asyncio
+from executors.connectors import Transfer, connectors
+from executors.connectors.baseconnector import BaseStorageConnector
+from executors.connectors.exceptions import DataTransferError
+from executors.socket_utils import BaseCommunicator, BaseProtocol, Message, PeerIdentity
+from executors.zeromq_utils import ZMQCommunicator
+
+from .transfer import transfer_data
+
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.DEBUG,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+logger = logging.getLogger(__name__)
+
+
+LISTENER_IP = os.getenv("LISTENER_IP", "127.0.0.1")
+LISTENER_PORT = os.getenv("LISTENER_PORT", "53893")
+LISTENER_PROTOCOL = os.getenv("LISTENER_PROTOCOL", "tcp")
+
+DATA_ID = int(os.getenv("DATA_ID", "-1"))
+
+DATA_LOCAL_VOLUME = Path(os.environ.get("DATA_LOCAL_VOLUME", "/data_local"))
+DATA_ALL_VOLUME = Path(os.environ.get("DATA_LOCAL_VOLUME", "/data_local"))
+INPUTS_VOLUME = Path(os.environ.get("INPUTS_VOLUME", "/inputs"))
+
+GENIALIS_UID = int(os.environ.get("GENIALIS_UID", 0))
+GENIALIS_GID = int(os.environ.get("GENIALIS_GID", 0))
+
+LOCATION_SUBPATH = Path(os.environ["LOCATION_SUBPATH"])  # No sensible default.
+
+DOWNLOAD_WAITING_TIMEOUT = 60  # in seconds
+RETRIES = 5
+
+DATA_ALL_VOLUME_SHARED = bool(
+    strtobool(os.environ.get("DATA_ALL_VOLUME_SHARED", "False"))
+)
+SET_PERMISSIONS = bool(strtobool(os.environ.get("INIT_SET_PERMISSIONS", "False")))
+
+
+async def transfer_inputs(communicator: BaseCommunicator):
+    """Transfer missing input data.
+
+    :raises DataTransferError: on failure.
+    """
+    inputs_connector = connectors["local"].duplicate()
+    inputs_connector.path = INPUTS_VOLUME
+    inputs_connector.name = "inputs"
+    inputs_connector.config["path"] = INPUTS_VOLUME
+
+    logger.debug(f"Inputs connector: {inputs_connector}")
+
+    logger.debug("Transfering missing data.")
+    response = await communicator.send_command(
+        Message.command("get_inputs_kubernetes", "")
```

Comment:
It is actually used when there is no shared filesystem for the data files.  Since this coincides with kubernetes I have chosen such name. Maybe it is not the most appropriate but I can not think of a better one currently.